src/eth/ProxyCdp.js

```diff
@@ -230,15 +160,35 @@ export default class ProxyCdp {
       }
     }
 
+    const getId = async () => {
+      const txObj = this._transactionManager.getTransaction(
+        this._transactionObject
+      );
+      return new Promise(resolve => {
+        txObj.onMined(async () => {
+          let log;
+          switch (txObj.metadata.method) {
+            case 'createAndOpen':
+              log = txObj.receipt.logs[5];
+              break;
+            case 'createOpenLockAndDraw':
+              log = txObj.receipt.logs[5];
+              break;
+            case 'lockAndDraw':
+              log = txObj.receipt.logs[2];
+              break;
+            case 'open':
+              log = txObj.receipt.logs[2];
+          }
+
+          resolve(this._web3Service._web3.utils.hexToNumber(log.data));
+        });
+      });
+    };
+
     const promise = (async () => {
-      // this "no-op await" is necessary for the inner reference to the
-      // outer promise to become valid
-      await 0;
-      const results = await Promise.all([
-        this._getCdpId(saiProxy.address, tub, dsProxyAddressPromise),
-        saiProxy[method](...args)
-      ]);
-      this.id = results[0];
+      this._transactionObject = saiProxy[method](...args);
```

Comment:
you can make the same change here as i described above, turning it into a local variable and passing it to getId as an argument